src/Domain/ContentRenderer/DelegatingContentRenderer.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\ExternalContent\Domain\ContentRenderer;
+
+use ProfessionalWiki\ExternalContent\Domain\ContentRenderer;
+
+class DelegatingContentRenderer implements ContentRenderer {
+
+	private ?ContentRenderer $renderer = null;
+
+	/**
+	 * @param string[] $arguments
+	 */
+	public function __construct(
+		private array $arguments
+	) {
+	}
+
+	public function render( string $content, string $contentUrl ): string {
+		$this->renderer = $this->createRenderer( $contentUrl );
```

Comment:
This bit feels a bit awkward, but since the renderer depends on the arguments and the URL I can only create the renderer here. The renderer is needed later (getOutputStyles and getOutputModuleStyles), hence the variable.

I'm not sure if this should be handled in a different way by moving the `$parser->getOutput()->...` elsewhere:
https://github.com/ProfessionalWiki/ExternalContent/pull/35/files#diff-7d2f76a8a218327ac9171ac3e294b57b3008237e2b8e4cebdfbfb940ada528efR32-R34

Reply:
Indeed awkward

Looks like we can get around this by having the logic that figures out what Renderer to use in the EmbedUseCase. A new service could be injected for that, replacing the currently injected ContentRenderer. Maybe ContentRendererFactory.

Injecting the unparsed parser function arguments into the constructor of this DelegatingRenderer also isn't great, since that is presentation layer stuff. Typically this would show up in the request model (parameters of the embed function on EmbedUseCase), already parsed.